### ISSUE TITLE
reclassify errors if `data_type=float`

### ISSUE THREAD

**Issue by echeipesh** [locationtech-labs/geopyspark#254]
The same call is succeeds with `data_type=int` 

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-12-47516ff0aa4d> in <module>()
      1 road_friction = (
      2     road_raster
----> 3     .reclassify({1:1}, data_type=float, replace_nodata_with=10)
      4 )

/home/hadoop/.local/lib/python3.4/site-packages/geopyspark/geotrellis/rdd.py in reclassify(self, value_map, data_type, boundary_strategy, replace_nodata_with)
    856         """
    857 
--> 858         srdd = _reclassify(self.srdd, value_map, data_type, boundary_strategy, replace_nodata_with)
    859 
    860         return TiledRasterRDD(self.geopysc, self.rdd_type, srdd)

/home/hadoop/.local/lib/python3.4/site-packages/geopyspark/geotrellis/rdd.py in _reclassify(srdd, value_map, data_type, boundary_strategy, replace_nodata_with)
     61         if not isinstance(key, data_type):
     62             val = value_map[key]
---> 63             for k in key:
     64                 new_dict[k] = val
     65         else:

TypeError: 'int' object is not iterable
```

**Comment by jbouffard** [comment]
The reason why this fails is because of this [check](https://github.com/locationtech-labs/geopyspark/blob/master/geopyspark/geotrellis/rdd.py#L36). The key of your `value_map` is an `int` and not a `float`, that check fails and it assumes that the key is instead a `list`. This isn't so much an error, as it is poor value checking. I'll have a PR up to resolve this soon.

**Comment by jbouffard** [comment]
Resolved in #264
